test/suite/test_las07.py

```diff
@@ -0,0 +1,142 @@
+        ds.populate()
+
+        uri2 = "table:las02_extra"
+        ds2 = SimpleDataSet(self, uri2, 0, key_format="S", value_format="S")
+        ds2.populate()
+
+        # Pin oldest and stable to timestamp 1.
+        self.conn.set_timestamp('oldest_timestamp=' + timestamp_str(1) +
+            ',stable_timestamp=' + timestamp_str(1))
+
+        bigvalue = "aaaaa" * 100
+        self.large_updates(uri, bigvalue, ds, nrows, 1)
+
+        # Check that all updates are seen
+        self.check(bigvalue, uri, nrows, 1)
+
+        # Check to see lookaside working with old timestamp
+        bigvalue2 = "ddddd" * 100
+        self.large_updates(uri, bigvalue2, ds, nrows, 100)
+
+        # Force out most of the pages by updating a different tree
+        self.large_updates(uri2, bigvalue, ds2, nrows, 100)
+
+        # Check that the new updates are only seen after the update timestamp
+        self.check(bigvalue, uri, nrows, 1)
+        self.check(bigvalue2, uri, nrows, 100)
+
+        # Pin oldest and stable to timestamp 100.
+        self.conn.set_timestamp('oldest_timestamp=' + timestamp_str(100) +
+            ',stable_timestamp=' + timestamp_str(100))
+
+        # Check that the new updates are only seen after the update timestamp
+        self.check(bigvalue2, uri, nrows, 100)
+
+        # Second set of update operations with increased timestamp
+        self.large_updates(uri, bigvalue, ds, nrows, 250)
+        self.large_updates(uri, bigvalue2, ds, nrows, 300)
+
+        # Force out most of the pages by updating a different tree
+        self.large_updates(uri2, bigvalue, ds2, nrows, 300)
+
+        # Check that the new updates are only seen after the update timestamp
+        self.check(bigvalue, uri, nrows, 250)
+        self.check(bigvalue2, uri, nrows, 300)
+
+        # Pin oldest and stable to timestamp 300.
+        self.conn.set_timestamp('oldest_timestamp=' + timestamp_str(300) +
+            ',stable_timestamp=' + timestamp_str(300))
+
+        # Check that the new updates are only seen after the update timestamp
+        self.check(bigvalue2, uri, nrows, 300)
+
+        # Third set of update operations with increased timestamp
+        self.large_updates(uri, bigvalue, ds, nrows, 350)
+        self.large_updates(uri, bigvalue2, ds, nrows, 400)
+
+        # Force out most of the pages by updating a different tree
+        self.large_updates(uri2, bigvalue, ds2, nrows, 400)
+
+        # Check that the new updates are only seen after the update timestamp
+        self.check(bigvalue, uri, nrows, 350)
+        self.check(bigvalue2, uri, nrows, 400)
+
+        # Pin oldest and stable to timestamp 400.
+        self.conn.set_timestamp('oldest_timestamp=' + timestamp_str(400) +
+            ',stable_timestamp=' + timestamp_str(400))
+
+        # Check that the new updates are only seen after the update timestamp
+        self.check(bigvalue2, uri, nrows, 400)
```

Comment:
This doesn't check that sweep is actually removing entries, does it? Is that possible to test?

Reply:
It's historically been hard to test that without introducing noise in the test suite.

Reply:
I modified the test to let the sweep server to get triggered during the running of the test, but it is difficult to verify it. Adding stats of how many records present in the LAS and the way to trigger the LAS sweep from test capability can provide the verification.